### PR TITLE
Bug 1821295 - Embed videos in Treeherder job Performance tab.

### PR DESCRIPTION
**Pull request by alexandru-io**

### CODE REVIEW

**Comment by alexandru-io** [comment]
image

image

**Comment by alexandru-io** [comment]
Screenshot from 2023-02-16 14-51-48

File: ui/job-view/details/tabs/SideBySide.jsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    };
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  componentDidUpdate(prevProps) {
+    this.getSideBySideParams();
+  }
+
+  getSideBySideParams() {
+    const { jobDetails } = this.state;
```
Comment (esanuandra) [inline ui/job-view/details/tabs/SideBySide.jsx:31]: I believe `jobDetails` should be from props.

**Comment by alexandru-io** [comment]
Screenshot from 2023-02-17 11-35-36
---
Screenshot from 2023-02-17 11-36-03

File: ui/job-view/details/tabs/SideBySide.jsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    //   warm: false,
+    //   most_similar: false,
+    //   search_crons: false,
+    //   skip_download: false,
+    //   output: null,
+    //   metric: 'speedindex',
+    //   vismetPath: false,
+    //   original: false,
+    //   skip_slow_gif: false,
+    // }
```
Comment (esanuandra) [inline ui/job-view/details/tabs/SideBySide.jsx:58]: do we still need these?

Comment (alexandru-io) [reply ui/job-view/details/tabs/SideBySide.jsx:58]: I'd like to keep it

Comment (esanuandra) [reply ui/job-view/details/tabs/SideBySide.jsx:58]: Maybe we could leave an explanation comment on why we need to keep it

File: ui/job-view/details/tabs/SideBySideVideo.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    this.setState({ activeVideo: { url: inactiveVideo } }, () =>
+      this.setState({
+        activeVideo: videoToReplay,
+        videoToReplay,
+        inactiveVideo: activeVideo,
+      }),
+    );
+  };
+
+  onSetVideo = (value) => {
```
Comment (esanuandra) [inline ui/job-view/details/tabs/SideBySideVideo.jsx:30]: Could be better to change the name in `onSetVideoHandler`.

**Review by esanuandra — APPROVED** [review]
Nice work!

**Pull request merged**